linkerd/tls/src/client.rs

```diff
@@ -38,11 +46,16 @@ pub enum NoClientTls {
     NotProvidedByServiceDiscovery,
 }
 
-/// Indicates whether the target server endpoint has a known TLS identity.
-pub type ConditionalClientTls = Conditional<ServerId, NoClientTls>;
+/// A stack paramater that indicates whether the target server endpoint has a
+/// known TLS identity.
+pub type ConditionalClientTls = Conditional<ClientTls, NoClientTls>;
 
 pub type Config = Arc<rustls::ClientConfig>;
 
+/// A stack param that configures the available ALPN protocols.
+#[derive(Clone, Debug, Eq, PartialEq, Hash)]
+pub struct AlpnProtocols(pub Vec<Vec<u8>>);
```

Comment:
I guess the `Vec` representation is locked in due to the `rustls` API? It would be nice to not have to clone the bytes around constantly. But, it's probably not big enough to be terribly important.